plugin/torch/torch_criterion-inl.h

```diff
@@ -61,7 +60,7 @@ class TorchCriterionOp : public Operator {
     int err = lua_pcall(L, 0, 1, 0);
     CHECK_EQ(err, 0) << lua_tostring(L, -1);
     // serialize
-    TorchState::Serialize(&chunk_);
+    this->lua_reference_ = lua_ref(L, LUA_REGISTRYINDEX);
```

Comment:
I think this is also ok as long as no internals of torch layer object will be garbage collected.

Since torch plugin is not in test cases, could you run the demo in exmaples/torch/torch_module/criterion to verify that this is working?